be/src/io/fs/file_writer.h

```diff
@@ -54,25 +55,26 @@ class FileWriter {
     FileWriter(const FileWriter&) = delete;
     const FileWriter& operator=(const FileWriter&) = delete;
 
-    // Normal close. Wait for all data to persist before returning.
-    // If there is no data appended, an empty file will be persisted.
-    virtual Status close() = 0;
+    Status close(bool non_block = false);
```

Comment:
usually we dont use doubly "negative" to present default value,
better change the signature to 
```
    Status close(bool blocking = true);
```

Reply:
also add sufficient comment for this interface, behavior and params

Reply:
we may need UT for the async/nonblocking test